src/dialect/mod.rs

```diff
@@ -880,6 +880,15 @@ pub trait Dialect: Debug + Any {
     fn supports_table_hints(&self) -> bool {
         false
     }
+
+    /// Returns whether it's the start of a single line comment
+    /// e.g. MySQL requires a space after `--` to be a single line comment
+    /// Otherwise it's interpreted as a double minus operator
+    ///
+    /// MySQL: <https://dev.mysql.com/doc/refman/8.4/en/ansi-diff-comments.html>
+    fn requires_whitespace_to_start_comment(&self) -> bool {
```

Comment:
```suggestion
    fn requires_single_line_comment_whitespace(&self) -> bool {
```
thinking something like this to flag that this is only for the `--` comment style